src/registry/artifactory.rs

```diff
@@ -40,6 +40,36 @@ impl Artifactory {
 impl Registry for Artifactory {
     /// Downloads a package from artifactory
     async fn download(&self, dependency: Dependency) -> eyre::Result<Package> {
+        ensure!(
+            dependency.manifest.version.comparators.len() == 1,
+            "{} uses unsupported semver comparators",
+            dependency.package
+        );
+
+        let version = dependency
+            .manifest
+            .version
+            .comparators
+            .first()
+            .wrap_err("internal error")?;
+
+        ensure!(
+            version.op == semver::Op::Exact && version.minor.is_some() && version.patch.is_some(),
+            "artifactory only support pinned dependencies"
+        );
```

Comment:
It might be nicer to move this validation where the version requirement is added. But I guess this will change soon so it's not a big deal.

Reply:
Well, the requirements are valid in the manifest they just can't be used with the artifactory registry impl